src/utils/expensesFilter.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const expensesService = require('../services/expenses.services');
+
+function isDateAfter(dateToCheck, referenceDate) {
+  const dateA = new Date(dateToCheck).valueOf();
+  const dateB = new Date(referenceDate).valueOf();
+
+  return dateA > dateB;
+}
+
+function isDateBefore(dateToCheck, referenceDate) {
+  const dateA = new Date(dateToCheck).valueOf();
+  const dateB = new Date(referenceDate).valueOf();
+
+  return dateA < dateB;
+}
+
+const expensesFilter = (
+  userId,
+  from,
+  to,
+  categories
+) => {
+  let expenses = expensesService.getAll();
+
+  if (userId) {
+    expenses = expenses.filter(expense => expense.userId === +userId);
+  }
+
+  if (categories) {
+    expenses = expenses.filter(({ category }) => categories.includes(category));
```

Comment:
if there are no `categories` in search params we will get `undefined`
if only 1 `categories` we will get just value (....?categories=1)
And if we have many categories we will get array of values (....?categories=1&categories=2&categories=3)

![image](https://github.com/mate-academy/node_accounting-app-v2/assets/86789161/ea010564-92a2-4116-9fcf-a66739d6d01f)

Your code will crash if you have only 1 category